src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsWindowMapping.java

```diff
@@ -69,6 +94,67 @@ public static LocalDateTime getFirstDateTimeMustBe() {
         return null;
     }
 
+    @NotNull
+    private static LocalDate handleAdjournedHearingType(SscsCaseData caseData) {
+        String dateType = caseData.getAdjournCaseNextHearingDateType();
+        if (FIRST_AVAILABLE_DATE.equalsIgnoreCase(dateType)) {
+            return LocalDate.now().plusDays(DAYS_TO_ADD_HEARING_WINDOW_TODAY_ADJOURNMENT);
+
+        } else if (FIRST_AVAILABLE_DATE_AFTER.equalsIgnoreCase(dateType)) {
+            return calculateDateOfAdjournedHearing(caseData);
+
+        } else {
+            throw new IllegalArgumentException("Unexpected adjournCaseNextHearingDateType: "
+                 + dateType);
+        }
+    }
+
+    private static LocalDate calculateDateOfAdjournedHearing(SscsCaseData caseData) {
+        String dateOrPeriod = caseData.getAdjournCaseNextHearingDateOrPeriod();
+
+        checkForMissingData(
+            dateOrPeriod,
+            "adjournCaseNextHearingDateOrPeriod"
+        );
+
+        if (PROVIDE_PERIOD.equalsIgnoreCase(dateOrPeriod)) {
+            checkForMissingData(
+                caseData.getAdjournCaseNextHearingFirstAvailableDateAfterPeriod(),
+                "adjournCaseNextHearingFirstAvailableDateAfterPeriod"
+            );
+
+            return parseDaysFromPeriod(caseData);
+
+        } else if (PROVIDE_DATE.equalsIgnoreCase(dateOrPeriod)) {
+            checkForMissingData(
+                caseData.getAdjournCaseNextHearingFirstAvailableDateAfterDate(),
+                "adjournCaseNextHearingFirstAvailableDateAfterDate"
+            );
+
+            return LocalDate.parse(caseData.getAdjournCaseNextHearingFirstAvailableDateAfterDate());
+
+        } else {
+            throw new IllegalArgumentException("Unexpected adjournCaseNextHearingDateOrPeriod: " + dateOrPeriod);
+        }
+    }
+
+    @NotNull
+    private static LocalDate parseDaysFromPeriod(SscsCaseData caseData) {
+        return LocalDate.now().plusDays(Integer.parseInt(
+            caseData.getAdjournCaseNextHearingFirstAvailableDateAfterPeriod().substring(0, 2)));
+    }
+
+    private static void checkForMissingData(String caseDataField, String missingField) {
+        if (isNull(caseDataField)) {
+            throw new IllegalArgumentException(missingField + " unexpectedly null");
```

Comment:
Can you use a NullPointerException

Reply:
I think NPE is not quite right to use here. IllegalArgumentException or IllegalStateException are correct. https://stackoverflow.com/questions/2517965/what-exception-to-throw-when-an-important-parameter-dependency-is-missing

Reply:
Agreed